### PR TITLE
Move network target support wrappers to their own header file

### PR DESCRIPTION
**Pull request by acass91**
This PR is to move the network target support wrappers to their own header file. The details are discussed in issue https://github.com/goatshriek/stumpless/issues/215

Output of `make check`:

```
      Start  1: journald
 1/61 Test  #1: journald .........................   Passed    0.07 sec
      Start  2: journald_supported
 2/61 Test  #2: journald_supported ...............   Passed    0.00 sec
      Start  3: network
 3/61 Test  #3: network ..........................   Passed    0.17 sec
      Start  4: tcp4
 4/61 Test  #4: tcp4 .............................   Passed    0.00 sec
      Start  5: tcp6
 5/61 Test  #5: tcp6 .............................   Passed    0.00 sec
      Start  6: udp4
 6/61 Test  #6: udp4 .............................   Passed    0.00 sec
      Start  7: udp6
 7/61 Test  #7: udp6 .............................   Passed    0.00 sec
      Start  8: network_leak
 8/61 Test  #8: network_leak .....................   Passed    0.00 sec
      Start  9: tcp4_leak
 9/61 Test  #9: tcp4_leak ........................   Passed    0.00 sec
      Start 10: tcp6_leak
10/61 Test #10: tcp6_leak ........................   Passed    0.00 sec
      Start 11: udp4_leak
11/61 Test #11: udp4_leak ........................   Passed    0.01 sec
      Start 12: udp6_leak
12/61 Test #12: udp6_leak ........................   Passed    0.00 sec
      Start 13: socket
13/61 Test #13: socket ...........................   Passed    0.01 sec
      Start 14: socket_add_malloc_failure
14/61 Test #14: socket_add_malloc_failure ........   Passed    0.00 sec
      Start 15: socket_supported
15/61 Test #15: socket_supported .................   Passed    0.00 sec
      Start 16: wel_unsupported
16/61 Test #16: wel_unsupported ..................   Passed    0.00 sec
      Start 17: buffer
17/61 Test #17: buffer ...........................   Passed    0.01 sec
      Start 18: buffer_leak
18/61 Test #18: buffer_leak ......................   Passed    0.00 sec
      Start 19: current_target
19/61 Test #19: current_target ...................   Passed    0.00 sec
      Start 20: element
20/61 Test #20: element ..........................   Passed    0.00 sec
      Start 21: element_leak
21/61 Test #21: element_leak .....................   Passed    0.00 sec
      Start 22: entry
22/61 Test #22: entry ............................   Passed    0.00 sec
      Start 23: entry_leak
23/61 Test #23: entry_leak .......................   Passed    0.00 sec
      Start 24: entry_memory_failure
24/61 Test #24: entry_memory_failure .............   Passed    0.00 sec
      Start 25: error
25/61 Test #25: error ............................   Passed    0.00 sec
      Start 26: error_leak
26/61 Test #26: error_leak .......................   Passed    0.00 sec
      Start 27: error_memory_failure
27/61 Test #27: error_memory_failure .............   Passed    0.00 sec
      Start 28: facility
28/61 Test #28: facility .........................   Passed    0.00 sec
      Start 29: file
29/61 Test #29: file .............................   Passed    0.01 sec
      Start 30: filter
30/61 Test #30: filter ...........................   Passed    0.00 sec
      Start 31: function
31/61 Test #31: function .........................   Passed    0.00 sec
      Start 32: get_error_stream
32/61 Test #32: get_error_stream .................   Passed    0.00 sec
      Start 33: level_all_disabled
33/61 Test #33: level_all_disabled ...............   Passed    0.00 sec
      Start 34: level_all_enabled
34/61 Test #34: level_all_enabled ................   Passed    0.00 sec
      Start 35: level_disabled_downto_alert
35/61 Test #35: level_disabled_downto_alert ......   Passed    0.00 sec
      Start 36: level_disabled_downto_emerg
36/61 Test #36: level_disabled_downto_emerg ......   Passed    0.00 sec
      Start 37: level_disabled_downto_err
37/61 Test #37: level_disabled_downto_err ........   Passed    0.00 sec
      Start 38: level_disabled_downto_crit
38/61 Test #38: level_disabled_downto_crit .......   Passed    0.00 sec
      Start 39: level_disabled_downto_debug
39/61 Test #39: level_disabled_downto_debug ......   Passed    0.00 sec
      Start 40: level_disabled_downto_info
40/61 Test #40: level_disabled_downto_info .......   Passed    0.00 sec
      Start 41: level_disabled_downto_notice
41/61 Test #41: level_disabled_downto_notice .....   Passed    0.01 sec
      Start 42: level_disabled_downto_trace
42/61 Test #42: level_disabled_downto_trace ......   Passed    0.00 sec
      Start 43: level_disabled_downto_warning
43/61 Test #43: level_disabled_downto_warning ....   Passed    0.00 sec
      Start 44: level_enable_upto_alert
44/61 Test #44: level_enable_upto_alert ..........   Passed    0.00 sec
      Start 45: level_enable_upto_crit
45/61 Test #45: level_enable_upto_crit ...........   Passed    0.00 sec
      Start 46: level_enable_upto_debug
46/61 Test #46: level_enable_upto_debug ..........   Passed    0.00 sec
      Start 47: level_enable_upto_emerg
47/61 Test #47: level_enable_upto_emerg ..........   Passed    0.00 sec
      Start 48: level_enable_upto_err
48/61 Test #48: level_enable_upto_err ............   Passed    0.00 sec
      Start 49: level_enable_upto_info
49/61 Test #49: level_enable_upto_info ...........   Passed    0.00 sec
      Start 50: level_enable_upto_notice
50/61 Test #50: level_enable_upto_notice .........   Passed    0.00 sec
      Start 51: level_enable_upto_trace
51/61 Test #51: level_enable_upto_trace ..........   Passed    0.00 sec
      Start 52: level_enable_upto_warning
52/61 Test #52: level_enable_upto_warning ........   Passed    0.00 sec
      Start 53: log
53/61 Test #53: log ..............................   Passed    0.04 sec
      Start 54: memory
54/61 Test #54: memory ...........................   Passed    0.00 sec
      Start 55: param
55/61 Test #55: param ............................   Passed    0.00 sec
      Start 56: perror
56/61 Test #56: perror ...........................   Passed    0.00 sec
      Start 57: severity
57/61 Test #57: severity .........................   Passed    0.00 sec
      Start 58: stream
58/61 Test #58: stream ...........................   Passed    0.01 sec
      Start 59: target
59/61 Test #59: target ...........................   Passed    0.04 sec
      Start 60: target_leak
60/61 Test #60: target_leak ......................   Passed    0.00 sec
      Start 61: version
61/61 Test #61: version ..........................   Passed    0.00 sec

100% tests passed, 0 tests failed out of 61

Total Test time (real) =   0.55 sec
[100%] Built target check

```
Output of `tools/check_headers/check_headers.rb "include/**/*.h*" "src/**/*.c" "test/**/*.cpp" "docs/examples/**/*.c"`:
```
warning: deprecated term stumpless_destroy_element used in include/stumpless/element.h
warning: deprecated term stumpless_destroy_entry used in include/stumpless/entry.h
warning: deprecated term stumpless_destroy_element used in src/element.c
warning: deprecated term stumpless_destroy_entry used in src/entry.c
```

### CODE REVIEW

**Comment by acass91** [comment]
@goatshriek Before I move this PR out of draft, does this look okay?

**Comment by goatshriek** [comment]
Yes, it looks good! It looks like the network targets supported define is missing from the config.h manifest yaml file, which is causing the static analysis failure. If you add that in it should resolve the failure.

File: include/private/config/network_support_wrapper.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+/*
+ * Copyright 2021 Joel E. Anderson
```
Comment (goatshriek) [inline include/private/config/network_support_wrapper.h]: update this to 2022

File: include/private/config/network_support_wrapper.h [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __STUMPLESS_PRIVATE_CONFIG_WRAPPER_NETWORK_SUPPORT_H
+#  define __STUMPLESS_PRIVATE_CONFIG_WRAPPER_NETWORK_SUPPORT_H
+
+#  include <stumpless/config.h>
+
```
Comment (goatshriek) [inline include/private/config/network_support_wrapper.h:89]: You'll also need to move the preprocessor inclusion of the windows wrapper, as well as the `HAVE_SYS_SOCKET`/`HAVE_WINSOCK2` blocks, starting from line 25 in the original `include/private/config/wrapper.h` file to line 87.

**Review by goatshriek — CHANGES_REQUESTED** [review]
Just a few more changes to make before this can get merged in. There is another block of network-specific inclusions and defines that need to be moved as well as part of this change, in order to complete the separation of the network definitions from the general ones.

File: include/private/config/network_support_wrapper.h (hunk 2 above)
Comment (acass91) [reply include/private/config/network_support_wrapper.h:89]: Can do.

File: include/private/config/network_support_wrapper.h (hunk 1 above)
Comment (acass91) [reply include/private/config/network_support_wrapper.h]: Updated.

**Comment by goatshriek** [comment]
Thanks very much for putting this together!

**Pull request merged**